### PR TITLE
Redocument Fiber#transfer

### PR DESCRIPTION
**Pull request by zverok**
I tried to document and showcase [new `Fiber#transfer` limitations](https://bugs.ruby-lang.org/issues/17221) to the best of my understanding, so it would make a consistent example, not just a list of unrelated rules.

Open to suggestions.

### CODE REVIEW

File: cont.c [hunk 1]
```diff
@@ -2460,42 +2460,68 @@ rb_fiber_backtrace_locations(int argc, VALUE *argv, VALUE fiber)
...
- *  You cannot call +resume+ on a fiber that has been transferred to.
- *  If you call +transfer+ on a fiber, and later call +resume+ on the
- *  the fiber, a +FiberError+ will be raised. Once you call +transfer+ on
- *  a fiber, the only way to resume processing the fiber is to
- *  call +transfer+ on it again.
+ *  The two style of control passing to and from fiber (one is #resume and
+ *  Fiber::yield, another is #transfer to and from fiber) can't be freely
+ *  mixed.
+ *
+ *  * If the Fiber's lifecycle had started with transfer, it will never
```
Comment (ko1) [inline cont.c:2467]:
```ruby
Fiber.new{
  Fiber.new{}.resume
}.transfer
```

It works.

File: cont.c [hunk 2]
```diff
@@ -2460,42 +2460,68 @@ rb_fiber_backtrace_locations(int argc, VALUE *argv, VALUE fiber)
  *  a resume call. Arguments passed to transfer are treated like those
  *  passed to resume.
  *
- *  You cannot call +resume+ on a fiber that has been transferred to.
- *  If you call +transfer+ on a fiber, and later call +resume+ on the
- *  the fiber, a +FiberError+ will be raised. Once you call +transfer+ on
- *  a fiber, the only way to resume processing the fiber is to
- *  call +transfer+ on it again.
+ *  The two style of control passing to and from fiber (one is #resume and
```
Comment (ko1) [inline cont.c:2463]: It is helpful that `transfer` is advanced version and difficult to use correctly.
resume/yield pairs are managed version.
For example, the exception are propagated resuming fiber.

**Comment by ko1** [comment]
Thank you, I tried to comment some points.

**Comment by zverok** [comment]
@ko1 Thanks, I added/changed some text to address your concerns!

**Comment by ko1** [comment]
Thanks!

**Pull request merged**